Recipes/Recipes.API/Controllers/RecipesController.cs

```diff
@@ -38,4 +38,70 @@ public async Task<ActionResult<IEnumerable<Recipe>>> GetAllRecipes()
 
         return Ok(responseRecipes);
     }
+
+    [HttpGet("{recipeId:int}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<Recipe>> GetRecipe(int recipeId)
+    {
+        RecipeEntity? recipe;
+
+        try
+        {
+            recipe = await recipesRepository.GetByIdAsync(recipeId);
+        }
+        catch(Exception)
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError);
+        }
+
+        if(recipe == null)
+        {
+            return NotFound();
+        }
+
+        Recipe responseRecipe = new Recipe
```

Comment:
Seems like this is can be reused. Extract mapping from a RecipeEntity to Recipe as a separate method inside this controller class.